### PR TITLE
Reolling forward property-based tests for coders

### PR DESCRIPTION
**Pull request by pabloem**
I've tested these several times, to make sure that the test will remain consistently passing.

r: @Abacn 

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

File: sdks/python/apache_beam/coders/coders_property_based_test.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+# This strategy currently generates rows with simple types (i.e. non-list, and
+# non-map fields).
+SCHEMA_GENERATOR_STRATEGY = st.lists(
+    st.tuples(
+        st.text(ascii_letters + digits + '_', min_size=1),
+        st.sampled_from(SCHEMA_TYPES),
+        st.booleans()))
+
+FORBIDDEN_FIELD_NAMES = {
+    'as', 'for', 'in', 'with', 'or', 'is', 'not', 'if', 'elif', 'else'
```
Comment (Abacn) [inline sdks/python/apache_beam/coders/coders_property_based_test.py:90]: Thanks for the fix.

These are all python keywords but not all Python keywords. Is there any source why these names are fobidden, or there is a list of forbidden field names?

Comment (pabloem) [reply sdks/python/apache_beam/coders/coders_property_based_test.py]: thanks Yi! Great observation. I've added a keyword check instead, which is the same one used when building namedtuples, which is what row coders use underneath.

**Comment by TheNeuralBit** [comment]
FYI as a plane ride project I started looking at using a recursive rule to generate Beam schemas: https://github.com/apache/beam/pull/23361

You may be interested in the `field_names` strategy I added there.

Also, it would be great to combine these two efforts (first generate a schema, then generate test data given a schema) down the road!

**Comment by pabloem** [comment]
I created https://github.com/apache/beam/issues/23436 to track switching over - could we move forward with this for now? : )

**Review by Abacn — APPROVED** [review]
Thanks LGTM!

**Comment by TheNeuralBit** [comment]
> I created #23436 to track switching over - could we move forward with this for now? : )

Yep! Sorry I didn't mean to hold this up.

**Pull request merged**